### ISSUE TITLE
How to accses Outputs from custom callback with generator 

### ISSUE THREAD

**Issue by lysecret2** [rstudio/keras#305]
Greetings, 
I was following the example of the custom callback on the rstudio website and there you could access the stored list by history$losses. However with the same callback this is not possible when replacing fit by fit generator.
Thanks for your help

**Comment by jjallaire** [comment]
That might be just what is provided by the Keras generator interface.

Could you make sure that your callback has a reference to the model and access the history directly off of the model object?

**Comment by lysecret2** [comment]
Hi, so I managed to store the results this way: 
`                          self$model$history$history$AUC_Test <- c(self$model$history$history$AUC_Test, AUC_Test)`

**Comment by jjallaire** [comment]
Note that this is exactly where custom metrics would be stored. Would writing a custom metric be a more elegant solution? (https://keras.rstudio.com/reference/metric_binary_accuracy.html#custom-metrics)

**Comment by lysecret2** [comment]
Custom metrics are calculated per batch right? If so that isnt really an option. Any idea how to print?
